samples/apply-dictionary-renderer-to-graphics-overlay/src/main/java/com/esri/arcgismaps/sample/applydictionaryrenderertographicsoverlay/components/ApplyDictionaryRendererToGraphicsOverlayViewModel.kt

```diff
@@ -0,0 +1,252 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.applydictionaryrenderertographicsoverlay.components
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.geometry.Multipoint
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.geometry.SpatialReference
+import com.arcgismaps.mapping.ArcGISScene
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.PortalItem
+import com.arcgismaps.mapping.symbology.DictionaryRenderer
+import com.arcgismaps.mapping.symbology.DictionarySymbolStyle
+import com.arcgismaps.mapping.view.Camera
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.portal.Portal
+import com.arcgismaps.toolkit.geoviewcompose.SceneViewProxy
+import com.esri.arcgismaps.sample.applydictionaryrenderertographicsoverlay.R
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import org.xmlpull.v1.XmlPullParser
+import org.xmlpull.v1.XmlPullParserFactory
+import java.io.File
+
+/**
+ * ViewModel for the "Apply dictionary renderer to graphics overlay" sample.
+ */
+class ApplyDictionaryRendererToGraphicsOverlayViewModel(private val app: Application) : AndroidViewModel(app) {
+
+    // The scene shown in the SceneView composable
+    val arcGISScene = ArcGISScene(basemapStyle = BasemapStyle.ArcGISTopographic)
+
+    // Graphics overlay that will hold the message graphics
+    val graphicsOverlay: GraphicsOverlay = GraphicsOverlay()
+
+    // SceneViewProxy to enable programmatic viewpoint changes
+    val sceneViewProxy = SceneViewProxy()
+
+    // Used to display error messages
+    val messageDialogVM = MessageDialogViewModel()
+
+    // Provision path where downloaded sample assets will be placed by the downloader activity
+    private val provisionPath: String by lazy {
+        app.getExternalFilesDir(null)?.path.toString() + File.separator + app.getString(R.string.apply_dictionary_renderer_to_graphics_overlay_app_name)
+    }
+
+    init {
+        viewModelScope.launch(Dispatchers.IO) {
+            // Load the scene first
+            arcGISScene.load().onFailure { throwable ->
+                messageDialogVM.showMessageDialog(throwable)
+            }
+
+            // Create and apply dictionary renderer from a web style
+            val dictionaryRendererDeferred = viewModelScope.async(Dispatchers.IO) {
+                createMil2525dDictionaryRenderer()
+            }
```

Comment:
If the function `createMil2525dDictionaryRenderer` is updated to instead return a `Result<DictionaryRenderer>` you can avoid launching a new scope and await for the result via `getOrElse`. This way errors can now be abstracted within the init block:
```suggestion
            val dictionaryRenderer = createMil2525dDictionaryRenderer().getOrElse { throwable ->
                return@launch messageDialogVM.showMessageDialog(throwable)
            }
```